plugins/pulp_rpm/plugins/migrations/0016_new_yum_distributor.py

```diff
@@ -176,3 +177,30 @@ def _re_publish_repository(repo, distributor):
 def _remove_legacy_publish_dirs():
     _clear_orphaned_publish_dirs(OLD_ROOT_PUBLISH_DIR, OLD_HTTP_PUBLISH_DIR)
     _clear_orphaned_publish_dirs(OLD_ROOT_PUBLISH_DIR, OLD_HTTPS_PUBLISH_DIR)
+
+
+def distributor_working_dir(distributor_type_id, repo_id, mkdir=True):
+    """
+    Determines the working directory for a distributor to use for a repository.
+    If the mkdir argument is set to true, the directory will be created as
+    part of this call. See the module-level docstrings for more information on
+    the directory structure.
+    :param mkdir: if true, this call will create the directory; otherwise the
+                  full path will just be generated
+    :type  mkdir: bool
+    :return: full path on disk to the directory the distributor can use for the
+             given repository
+    :rtype:  str
+    """
+    storage_dir = pulp_config.config.get('server', 'storage_dir')
+    repo_working_dir = os.path.join(storage_dir, 'working', 'repos', repo_id)
+
+    if mkdir and not os.path.exists(repo_working_dir):
+        os.makedirs(repo_working_dir)
+
+    working_dir = os.path.join(repo_working_dir, 'distributors', distributor_type_id)
+
+    if mkdir and not os.path.exists(working_dir):
+        os.makedirs(working_dir)
```

Comment:
Since ```os.makedirs()``` is recursive, you could eliminate the first block of "if doesn't exist…" and just have this second block. You can also eliminate the repo_working_dir and just make a single expression for working_dir. If you make those two changes, this function is pretty simple!